autowarefoundation/autoware.universe
# feat(pose_initializer): check error initial pose and gnss pose, output diagnostics

## Description
- Check 2D(x-y) Error between Initial Pose result and GNSS Pose, if pose error is large, output an message in the /diagnostics topic.
- This feature is off by default, but can be enabled by setting pose_error_check_enabled to true when the GNSS is reliable, such as RTK-GNSS.

![Screenshot from 2024-09-24 18-48-16](https://github.com/user-attachments/assets/a01e93b0-bfe0-421d-a480-0c7980bd7ac1)

## Related links
https://github.com/autowarefoundation/autoware_launch/pull/1180

**Parent Issue:**

- Link



## How was this PR tested?
- checkout autoware_launch to [feat/initial_pose_check_with_gnss](https://github.com/autowarefoundation/autoware_launch/pull/1180)
- Check that localization works correctly in logging_simulator.
- Set pose_error_check_enabled to True and check that diagnostics (gnss_pose_error_2d, is_gnss_pose_error) are output. If set to False, gnss_pose_error_2d and is_gnss_pose_error are not output.

## Notes for reviewers

None.

## Interface changes

None.



## Effects on system behavior

None.
